### PR TITLE
throttled: new, 0.9.2

### PR DESCRIPTION
**Pull request by CamberLoid**
Topic Description
-----------------

Add package `throttled` in response to [pakreq 1008](https://pakreq.aosc.io/detail/1008)

Package(s) Affected
-------------------

`throttled` v0.9.2

Security Update?
----------------

No

Test Build(s) Done
------------------

    

- [ ] Architecture-independent `noarch`

Update(s) Uploaded to Stable
----------------------------

- [ ] Architecture-independent `noarch`

### CODE REVIEW

File: extra-admin/throttled/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,26 @@
+abinfo "Installing all scripts and configs ..."
+mkdir "$PKGDIR"/usr/lib/python"$ABPY3VER"/site-package/throttled -p
```
Comment (WhiredPlanck) [inline extra-admin/throttled/autobuild/build]: Every command needs to add verbosity if possible. For this case:

```suggestion
mkdir -pv "$PKGDIR"/usr/lib/python"$ABPY3VER"/site-package/throttled
```
...

File: extra-admin/throttled/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+install -Dvm755 "$SRCDIR"/throttled.py \
+    "$PKGDIR"/usr/lib/python"$ABPY3VER"/site-package/throttled/throttled.py
+install -Dvm755 "$SRCDIR"/mmio.py \
+    "$PKGDIR"/usr/lib/python"$ABPY3VER"/site-package/throttled/mmio.py
+install -Dvm644 "$SRCDIR"/etc/lenovo_fix.conf "$PKGDIR"/etc/throttled.conf
+install -Dvm644 "$SRCDIR"/systemd/lenovo_fix.service "$PKGDIR"/usr/lib/systemd/system/throttled.service
```
Comment (WhiredPlanck) [inline extra-admin/throttled/autobuild/build]: And for this case, you added verbosity, very good. But it's recommended that if the character numbers of a line is bigger than 80, insert a `\` into a proper place:

```suggestion
install -Dvm644 "$SRCDIR"/systemd/lenovo_fix.service \
    "$PKGDIR"/usr/lib/systemd/system/throttled.service
```

just like what you did at line 3, 4 and 5, 6.

File: extra-admin/throttled/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# in commit 392f3f8, for compability reasons, the config file is named to 
+# `throttled.conf` and the legacy config file `lenovo_fix.conf` will be a 
+# symbolic link to `throttled.conf`.
+ln -s throttled.conf "$PKGDIR"/etc/lenovo_fix.conf
+ln -s throttled.py "$PKGDIR"/usr/lib/python"$ABPY3VER"/site-package/throttled/lenovo_fix.py
+
+abinfo "Copying all pre-compiled pycache ..."
+cp -a "$SRCDIR"/__pycache__ "$PKGDIR"/usr/lib/python"$ABPY3VER"/site-package/throttled/
+
+abinfo "All done"
```
Comment (WhiredPlanck) [inline extra-admin/throttled/autobuild/build]: `abinfo`  at this line is unnecessary. 

Always add a blank line at the end of each file.

File: extra-admin/throttled/spec [hunk 4]
```diff
@@ -0,0 +1,5 @@
+VER=0.9.2
+# REL=0
```
Comment (WhiredPlanck) [inline extra-admin/throttled/spec]: If a variable is unused, just remove it.

File: extra-admin/throttled/autobuild/prepare [hunk 5]
```diff
@@ -0,0 +1,7 @@
+abinfo "Modifying systemd service file..."
+# Need update if upstream releases new package
+sed -i "s|ExecStart=.*|ExecStart=/bin/python3 /usr/lib/python$ABPY3VER/site-package/throttled.py|" "$SRCDIR"/systemd/lenovo_fix.service
```
Comment (WhiredPlanck) [inline extra-admin/throttled/autobuild/prepare]: For `sed` actions, we need to write them like this, which may be more clear to see what happened:

```suggestion
sed -e "s|ExecStart=.*|ExecStart=/bin/python3 /usr/lib/python$ABPY3VER/site-package/throttled.py|" \
    -i "$SRCDIR"/systemd/lenovo_fix.service
```

And here wouldn't it be more appropriate to create a symlink at `/usr/bin` to site-package path?

File: extra-admin/throttled/autobuild/defines [hunk 6]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=throttled
+PKGSEC=admin
+PKGDEP="dbus-python pygobject-3 python-3"
+
+PKGDES="Workaround for Intel throttling issues in Linux."
+
+ABHOST=noarch
```
Comment (WhiredPlanck) [inline extra-admin/throttled/autobuild/defines]: Add a blank line at the end.

File: extra-admin/throttled/autobuild/build [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+install -Dvm644 "$SRCDIR"/LICENSE "$PKGDIR"/usr/share/doc/"$PKGNAME"/LICENSE
+install -Dvm644 "$SRCDIR"/README.md "$PKGDIR"/usr/share/doc/"$PKGNAME"/README.md
+
+
+abinfo "Creating symbolic link ..."
+# As the project name is changed to `throttled` from the legacy name `lenovo_fix` 
+# in commit 392f3f8, for compability reasons, the config file is named to 
+# `throttled.conf` and the legacy config file `lenovo_fix.conf` will be a 
+# symbolic link to `throttled.conf`.
+ln -s throttled.conf "$PKGDIR"/etc/lenovo_fix.conf
```
Comment (WhiredPlanck) [inline extra-admin/throttled/autobuild/build]: Add verbosity: `ln -sv` ...

File: extra-admin/throttled/autobuild/prepare [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+abinfo "Pre-compiling all python scripts ..."
+mv lenovo_fix.py throttled.py
```
Comment (WhiredPlanck) [inline extra-admin/throttled/autobuild/prepare]: Add verbosity: `mv -v` ...

File: extra-admin/throttled/spec [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+
+SRCS="tbl::https://github.com/erpalma/throttled/archive/refs/tags/v$VER.tar.gz"
+CHKSUMS="sha256::01beb3e17303b11d86aeadc74d85d0ee2d50375998137dcdb5e464de4fdc4e07"
```
Comment (WhiredPlanck) [inline extra-admin/throttled/spec:4]: Here we need to add a variable called `CHKUPDATE`, which can help us to check if the new version of application is available. You could see [here](https://github.com/AOSC-Dev/aosc-findupdate/blob/master/docs/config.zh-CN.md) for more information.

**Review by WhiredPlanck — CHANGES_REQUESTED** [review]
Hi, @CamberLoid. Welcome to AOSC Community! And thanks your first pull request here ~ 

I've made several reviews below according our [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/). This can help us to make more better packages.

Best wishes!

**Comment by CamberLoid** [comment]
Thanks for mentoring.
Suggestion applied.

**Comment by CamberLoid** [comment]
duplicated.
closing now

**Pull request closed without merging**